lib/cylc/network/https/client_reporter.py

```diff
@@ -131,19 +131,22 @@ def signout(self, server_obj):
                 caller.user, caller.host, caller.prog_name, caller.uuid))
         try:
             del self.clients[caller.uuid]
-        except:
+        except KeyError:
             # Already forgotten.
             pass
         self._housekeep()
 
     def _housekeep(self):
         """Forget inactive clients."""
 
-        for uuid in self.clients.keys():
+        for uuid in self.clients.copy():
```

Comment:
Have fixed so that this makes sense.
An entry from the dictionary is being deleted whilst this loop is in progress, making a copy ensures that the contents don't change during the loop.